windows/installer-source/KaliteSetupScript.iss

```diff
@@ -242,14 +242,42 @@ begin
     end;
 end;
 
-procedure HandleUpgrade(targetPath : String);
+procedure SelectContentFolder;
 var
-    prevVerStr : String;
     userKaliteContent: String;
     systemKaliteDir: String;
     userPath : String;
     contentPath : String;
-    contentDb : String;
+    retCode: Integer;
+begin
+    userPath := ExpandConstant('{%USERPROFILE}')
+    contentPath := ExpandConstant('{%CONTENT_ROOT}')
+    systemKaliteDir := ExpandConstant(userPath + '\.kalite')
+    userKaliteContent := ExpandConstant(systemKaliteDir + '\content');
+    if Not DirExists(userKaliteContent) then
+    begin
+     if Not DirExists(contentPath) then
+        begin

```

Comment:
Oh you're right, the text only really fits in that case.

> This will always ask the user set there content path during installation but it's optional.

IMO it's fine to ask the user as long, as *if they are confused*, they will find it easy to just press cancel.

Let's try generalizing it some more:

```
'KA Lite Setup will look for video contents in the folder ' + contentPath + '. You can download contents to the selected folder after the installation completes - or - select a folder with already downloaded contents.\n\nIf you don't have a folder with downloaded contents or want to use the suggested folder, press Cancel.\n\nTo change the default folder for downloaded contents, press OK.'
```

Is that better?